### ISSUE TITLE
1.0-rc-3: IDEA chooses wrong overload for KClass<> with configuration overload for polymorphic containers

### ISSUE THREAD

**Issue by mkobit** [gradle/kotlin-dsl#1046]
Gradle works on the command line but the error is still highlighted in IntelliJ
### Expected Behavior

Given the following build script

```kotlin
open class SomeExtension

extensions.create("extension", SomeExtension::class)

tasks {
  register("wrapper", Wrapper::class).configure {
    gradleVersion = "4.10-rc-2"
  }

  register("1a", JavaCompile::class.java)
  register("1b", JavaCompile::class)

  register("2a", JavaCompile::class.java, Action {
  })
  register("2b", JavaCompile::class, Action {
  })

  register("3a", JavaCompile::class.java) {
  }
  register("3b", JavaCompile::class) {
  }
}
```

### Current Behavior

Running `./gradlew` compiles the script fine, but IDEA shows a compiler error.

![build.gradle.kts error highlighting from IDEA](https://user-images.githubusercontent.com/1447148/44215406-09d51480-a138-11e8-954a-995ffa64a063.png)

![IDEA suggesting different type](https://user-images.githubusercontent.com/1447148/44215408-0b9ed800-a138-11e8-874d-5f719aa4ed0e.png)

### Context

Confusing from an  "editing the build script" perspective.

### Steps to Reproduce (for bugs)

See script above and IDEA environment below

### Your Environment

Gradle 4.10-rc-2

IntelliJ IDEA 2018.2.1 (Ultimate Edition)
Build #IU-182.3911.36, built on August 6, 2018
Licensed to Mike Kobit
You have a perpetual fallback license for this version
Subscription is active until August 7, 2019
JRE: 1.8.0_152-release-1248-b8 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.15.0-32-generic

Kotlin Plugin - Version: 1.2.60-release-IJ2018.2-1

**Comment by eskatos** [comment]
Thanks for the complete report Mike!
I can reproduce

**Comment by bamboo** [comment]
This is probably caused by #1064 which is already fixed in Kotlin `1.2.70-dev-1738`:

<img width="970" alt="screen shot 2018-08-30 at 18 48 56" src="https://user-images.githubusercontent.com/51689/44881372-b7423f00-ac85-11e8-98a8-528c0a9fd275.png">

**Comment by eskatos** [comment]
Closing as duplicate of #1064
